### PR TITLE
Add geometry_msgs conversion functions

### LINKED ISSUE #117: Should provide common drake-type ROS geometry/eigen conversions
From glancing around Drake and Anzu, I'm not sure if I see a consolidated place for converting eigen/geometry messages to/from Drake types.

I think they should live in `drake_ros_core`, and we should start ingesting them in Anzu.

Conversions (ROS <-> C++)
- Point <-> `Eigen::Vector3d`
- Quaternion <-> `Eigen::Quaternion<double>`
- Pose, Transform <-> `Eigen::Isometry3d`, `drake::math::RigidTransformd`
- Twist <-> `Eigen::Vector6d`, `drake::multibody::SpatialVelocity`
- Acceleration <-> `Eigen::Vector6d`, `drake::multibody::SpatialAcceleration`
- Wrench <-> `Eigen::Vector6d`, `drake::multibody::SpatialForce`

For flavors of `Eigen::Vector6d`, please keep same conventions as Drake, that is `[rotational; linear]`

\cc @calderpg-tri @IanTheEngineer

### PR DESCRIPTION
**Pull request by EricCousineau-TRI**
Resolves #117
Successor to #141

- - -
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)

### CODE REVIEW

**Review by EricCousineau-TRI — COMMENTED** [review]
+@gbiggs and +@iantheengineer for review, please!

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-:-NEgr8RqAr0bvp-kCAcT:b-qj7b5v)* status: 0 of 5 files reviewed, all discussions resolved (waiting on @gbiggs and @IanTheEngineer)

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @EricCousineau-TRI, @gbiggs, and @IanTheEngineer)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhg00U4nxrsmbi4EIn:-NEhg00VBFdsqCzsDwKB:bbogr27) (no related file):*
Working: Ensure Bazel build works. (shouldn't be too much)

___
___
*[`drake_ros_core/include/drake_ros_core/geometry_conversions.h` line 131 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhg4U3B7JhRKuzrXKq:-NEhg4U47FKe40Lu5wJ3:b-ormbz5) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/include/drake_ros_core/geometry_conversions.h#L131)):*
> ```objc
>     const geometry_msgs::msg::Accel& accel);
> 
> geometry_msgs::msg::Accel SpatialAccelerationToRosAcceleration(
> ```

Working: Should `s/RosAccleration/RosAccel`, because message type is named `Accel`
(don't love the abbr., but whatevs)

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 0 of 5 files reviewed, all discussions resolved (waiting on @gbiggs and @IanTheEngineer)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhg00U4nxrsmbi4EIn:-NEhj292AaFK4-Haj8z8:b-896fix) (no related file):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Working: Ensure Bazel build works. (shouldn't be too much)
</blockquote></details>

Done.

___
___
*[`drake_ros_core/include/drake_ros_core/geometry_conversions.h` line 131 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhg4U3B7JhRKuzrXKq:-NEhj0fo3Kvjdyj_J5nn:b-896fix) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/include/drake_ros_core/geometry_conversions.h#L131)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Working: Should `s/RosAccleration/RosAccel`, because message type is named `Accel`
(don't love the abbr., but whatevs)
</blockquote></details>

Done.

**Review by IanTheEngineer — CHANGES_REQUESTED** [review]
Checkpoint. I reviewed both the Bazel and CMake buildfiles. I'll take another pass at this tomorrow for the conversion functions and tests.

> Reviewed 1 of 5 files at r1, 1 of 4 files at r2, 2 of 5 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-:-NEiL5ilEN_iFhytum2Y:b5nmps8)* status: 4 of 7 files reviewed, 3 unresolved discussions (waiting on @EricCousineau-TRI and @gbiggs)

___
*[`drake_ros_core/CMakeLists.txt` line 25 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhT76_2IuhJH0UHdER:-NEhT76_2IuhJH0UHdES:bjh4xl2) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/CMakeLists.txt#L25)):*
> ```CMake
> find_package(Eigen3 QUIET NO_MODULE)
> if(NOT Eigen3_FOUND)
>   find_package(Eigen3 REQUIRED)
> ```

There is a convenience package to handle Eigen3 CMake macros:
https://github.com/ros2/eigen3_cmake_module

That repo's README has details on how to set up the CMake invocation.
___
*[`drake_ros_core/package.xml` line 12 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhU-A5B7XHswf7tmHV:-NEhU-A630pKpmYVH4Fy:b-t1sdji) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/package.xml#L12)):*
> ```XML
> 
>   <buildtool_depend>ament_cmake_ros</buildtool_depend>
>   <buildtool_depend>eigen</buildtool_depend>
> ```

If we go the path of relying on the ```eigen_cmake_module```, then that package will need to be declared as a ```buildtool_depend```, and Eigen itself a ```build_depend```.



_Code quote:_
```XML
<buildtool_depend>eigen</buildtool_depend>
```
___
*[`drake_ros_core/package.xml` line 28 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhUTlgBJP1T4-4x7ho:-NEhUTlgBJP1T4-4x7hp:b-2k60o5) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/package.xml#L28)):*
> ```XML
> 
>   <export>
>     <build_type>ament_cmake</build_type>
> ```

Finally, ```eigen3_cmake_modules``` would need to be declared a ```buildtool_export_depend``` and ```eigen``` a ```build_export_depend``` as we use Eigen in the public headers.



_Code quote:_
```XML
<build_type>ament_cmake</build_type>
```

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 4 of 7 files reviewed, 3 unresolved discussions (waiting on @gbiggs and @IanTheEngineer)

___
*[`drake_ros_core/CMakeLists.txt` line 25 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhT76_2IuhJH0UHdER:-NEjbx2n5H5rxUtZawvx:blagq3h) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/CMakeLists.txt#L25)):*
<details><summary><i>Previously, IanTheEngineer (Ian McMahon) wrote…</i></summary><blockquote>

There is a convenience package to handle Eigen3 CMake macros:
https://github.com/ros2/eigen3_cmake_module

That repo's README has details on how to set up the CMake invocation.
</blockquote></details>

OK Thanks! However, I don't think you've stated a defect, nor do I think the defect should be blocking. Ultimately, I'd prefer to punt on this.
Can I leave it up to you to file as improvement?
___
*[`drake_ros_core/package.xml` line 12 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhU-A5B7XHswf7tmHV:-NEjc62F2Sx7AMqYuvmO:b2mrgyl) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/package.xml#L12)):*
<details><summary><i>Previously, IanTheEngineer (Ian McMahon) wrote…</i></summary><blockquote>

If we go the path of relying on the ```eigen_cmake_module```, then that package will need to be declared as a ```buildtool_depend```, and Eigen itself a ```build_depend```.
</blockquote></details>

OK See above.
___
*[`drake_ros_core/package.xml` line 28 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhUTlgBJP1T4-4x7ho:-NEjc8W6CMpR_R8dYj-5:b2mrgyl) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/package.xml#L28)):*
<details><summary><i>Previously, IanTheEngineer (Ian McMahon) wrote…</i></summary><blockquote>

Finally, ```eigen3_cmake_modules``` would need to be declared a ```buildtool_export_depend``` and ```eigen``` a ```build_export_depend``` as we use Eigen in the public headers.
</blockquote></details>

OK See above.

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 4 of 7 files reviewed, 4 unresolved discussions (waiting on @gbiggs and @IanTheEngineer)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEjcljSCoqb_3ByFzEx:-NEjcljSCoqb_3ByFzEy:beb70yw) (no related file):*
Working: CI is asking for lint fixes.

___

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 4 of 7 files reviewed, 3 unresolved discussions (waiting on @gbiggs and @IanTheEngineer)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEjcljSCoqb_3ByFzEx:-NEjfotC1HWmVqyN4NoD:b-896fix) (no related file):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Working: CI is asking for lint fixes.
</blockquote></details>

Done.

___

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 4 of 7 files reviewed, 4 unresolved discussions (waiting on @gbiggs and @IanTheEngineer)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEjvO1pAeIHppL0FYA9:-NEjvO1q3GWvTPGYqJ_T:bkhp2h1) (no related file):*
Working: Per #156, needs rebase to include #143.

___

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 1 of 7 files reviewed, 3 unresolved discussions (waiting on @gbiggs and @IanTheEngineer)

*[a discussion](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEjvO1pAeIHppL0FYA9:-NEjvYJa4Z4LJuc5hZ7Y:b-896fix) (no related file):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

Working: Per #156, needs rebase to include #143.
</blockquote></details>

Done.

___

**Review by IanTheEngineer — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 1 of 7 files reviewed, 3 unresolved discussions (waiting on @EricCousineau-TRI and @gbiggs)

___
*[`drake_ros_core/CMakeLists.txt` line 25 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhT76_2IuhJH0UHdER:-NEkWLxY56Pt0GWDd3ok:bheum96) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/CMakeLists.txt#L25)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

OK Thanks! However, I don't think you've stated a defect, nor do I think the defect should be blocking. Ultimately, I'd prefer to punt on this.
Can I leave it up to you to file as improvement?
</blockquote></details>

Apologies on not clearly stating the blocking defect: with the addition of Eigen headers included in the public API, downstream packages relying on `drake_ros_core` will fail to find the `Eigen3` headers at compile time through transitive dependencies. Eigen needs to be exported as a build export depend. The package I linked has CMake macros to mitigate this, and states best practices in their `README` for how to declare this in our `package.xml`. Per f2f convo, I'll verify this is still an issue with Focal's Eigen3.

**Review by EricCousineau-TRI — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153)* status: 0 of 7 files reviewed, 3 unresolved discussions (waiting on @gbiggs and @IanTheEngineer)

___
*[`drake_ros_core/CMakeLists.txt` line 25 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhT76_2IuhJH0UHdER:-NElOnao-dpxqZNwg1uB:b-4on7hk) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/CMakeLists.txt#L25)):*
<details><summary><i>Previously, IanTheEngineer (Ian McMahon) wrote…</i></summary><blockquote>

Apologies on not clearly stating the blocking defect: with the addition of Eigen headers included in the public API, downstream packages relying on `drake_ros_core` will fail to find the `Eigen3` headers at compile time through transitive dependencies. Eigen needs to be exported as a build export depend. The package I linked has CMake macros to mitigate this, and states best practices in their `README` for how to declare this in our `package.xml`. Per f2f convo, I'll verify this is still an issue with Focal's Eigen3.
</blockquote></details>

Done. Went ahead and added. Can remove if we find we don't need it.
___
*[`drake_ros_core/package.xml` line 12 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhU-A5B7XHswf7tmHV:-NElOrRy6_bY2U6z23mX:b-896fix) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/package.xml#L12)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

OK See above.
</blockquote></details>

Done.
___
*[`drake_ros_core/package.xml` line 28 at r2](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-NEhUTlgBJP1T4-4x7ho:-NElOsAU4j2qvKH84pAW:b-896fix) ([raw file](https://github.com/robotlocomotion/drake-ros/blob/cf6367ee9fea48371a502736e9cdf2e2ed5955ec/drake_ros_core/package.xml#L28)):*
<details><summary><i>Previously, EricCousineau-TRI (Eric Cousineau) wrote…</i></summary><blockquote>

OK See above.
</blockquote></details>

Done.

**Comment by EricCousineau-TRI** [comment]
I'd like to go ahead and have this ready to ingest into Anzu, and will go ahead and merge, and then fix-forward any bugs / defects that creep up. Thanks!

**Review by EricCousineau-TRI — COMMENTED** [review]
-@iantheengineer -@gbiggs +@ericcousineau-tri for self-stamp (given above statement)

> Reviewed 5 of 6 files at r6, 2 of 2 files at r7, all commit messages.
> Dismissed @IanTheEngineer from 3 discussions.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake-ros/153#-:-NErPfBM06N5X6z7y1V-:bvde8db)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @EricCousineau-TRI)

File: drake_ros_core/package.xml [hunk 1]
```diff
@@ -24,5 +32,7 @@
 
   <export>
     <build_type>ament_cmake</build_type>
+    <buildtool_export_depend>eigen3_cmake_module</buildtool_export_depend>
+    <build_export_depend>eigen</build_export_depend>
```
Comment (sloretz) [inline drake_ros_core/package.xml:36]: These lines actually belong on the same level as the other `<depend>` lines. They don't do anything inside an `<export>` block.

**Pull request merged**